packages/core/workers/src/cpuCount.js

```diff
@@ -1,66 +1,14 @@
 // @flow
 import os from 'os';
-import {execSync} from 'child_process';
 
-const exec = (command: string): string => {
-  try {
-    let stdout = execSync(command, {
-      encoding: 'utf8',
-      // This prevents the command from outputting to the console
-      stdio: [null, null, null],
-    });
-    return stdout.trim();
-  } catch (e) {
-    return '';
+let cachedCoreCount = 0;
+const platform = os.platform();
+export function getCoreCount(
+  limit: number = platform === 'win32' ? 4 : 8,
```

Comment:
This is not an OS-relevant issue. The parcel doesn't use the threads it spawns. Why do you want to create so many threads which Parcel doesn't use?

Reply:
I thought this was a filesystem limitation of windows

Reply:
That is a separate issue.

Reply:
The smarter way is to spawn threads proportional to the work that Parcel wants to do. For example, the number of the files.

```suggestion
  limit: number = 4,
```

Reply:
The smarter way is to spawn threads proportional to the work that Parcel wants to do. For example:

```js
max((fileNumbers)/10, coresAvailable)
```

Reply:
> The smarter way is to spawn threads proportional to the work that Parcel wants to do. For example:

Yes, but we don't always know this beforehand.

Reply:
We should use the information when it is available. When not known, we can just use a constant:

```js
fileNumbers ? min((fileNumbers)/20, coresAvailable) : 4
```

Reply:
What is `fileNumbers`?
On an initial build (where there's the most work), we don't know.
For subsequent builds, we only know the filecount of the previous build, not how much changed (= how much work there is).

So I would say this isn't possible to know at all

Reply:
So maybe it is better to perform an initial analysis using all threads. Then after that, it can use a suitable number of workers for the actual build.

Reply:
That would probably slow it down more than it would speed up?